### ISSUE TITLE
Document -export-debug option for exporting debug templates from command line

### ISSUE THREAD

**Issue by PkayJava** [godotengine/godot#5903]
Ubuntu 16.04 - 2.1.rc1.custom_build (latest commit)

I got the follow error log. and screen

ERROR: export_platform: Exporting failed!
   At: tools/editor/project_export.cpp:555.

![screenshot from 2016-07-26 00-02-32](https://cloud.githubusercontent.com/assets/872686/17110053/10e1275a-52c5-11e6-936b-40c43743ed3e.png)

**Steps to reproduce:**

just cd into a godot project and run the follow export command.
godot -export Android ../test.apk

it complain that it could not find certificate file.

but if I export manually from the export menu, it works.

**Comment by akien-mga** [comment]
I can confirm this (on Linux too), exporting to Android from the command line seems broken.

**Comment by akien-mga** [comment]
Actually the error already happened in 2.0.x; since the 2.1 release is imminent, moving to the next milestone. (A fix could be backported for 2.1.x releases of course).

**Comment by bojidar-bg** [comment]
What is the status here (esp. that 2.1.1 was released...)

**Comment by AlexHolly** [comment]
I had the same issue using the export command.

The android export command is using p_debug=false

https://github.com/godotengine/godot/blob/2.1/platform/android/export/export.cpp#L1325

so you need to set the release_keystore,pass and user in EditorSettings I also had to add this

```
if (release_keystore=="") {
	EditorSettings::get_singleton()->get("android/debug_keystore");
	keystore=EditorSettings::get_singleton()->get("android/release_keystore");
	password=EditorSettings::get_singleton()->get("android/release_keystore_pass");
	user=EditorSettings::get_singleton()->get("android/release_keystore_user");
} else {
	keystore=release_keystore;
	password=release_password;
	user=release_username;
}
````

Is there a way to pass the debug flag via command?

**Comment by akien-mga** [comment]
So if I understand correctly the issue is not with the command line export itself, but just that it exports by default in release mode and all of us who reproduced the issue here had not configured the release keystore?

**Comment by akien-mga** [comment]
So there is actually an undocumented `-export-debug` switch to use with `-export` and get the debug template: https://github.com/godotengine/godot/blob/5ae78fd/main/main.cpp#L1087

It should be added to the help output, cc @Calinou.

**Comment by DanielKinsman** [comment]
This may explain why I am having so much trouble exporting to android from command line via a docker, but just naively adding `-export-debug` doesn't seem to be helping. 

https://github.com/DanielKinsman/godot_server_docker

    docker build . --tag godot:latest 
    docker run -v /some/path:/mnt/working godot godot_server -path /mnt/working/game/ -export-debug -export "Android" /mnt/working/game.apk

Still fails to export. But if I vnc in to the docker container and start the godot gui, the export works.

**Comment by akien-mga** [comment]
It's properly documented in the master branch now, still needs to be added for 2.1. I haven't tested it though so I can't explain why it's not working for you @DanielKinsman.

**Comment by bojidar-bg** [comment]
It might be because it is `--export-debug` (with a double `-`) in 3.0, no?

**Comment by akien-mga** [comment]
Looking at the Dockerfile, it seems to me that @DanielKinsman is trying to build 2.1.4.

**Comment by DanielKinsman** [comment]
Ah it is `export_debug` *not* `export-debug` (underscore not hyphen). It would be handy if invalid command line args gave error messages, e.g. "unknown option -export-debug".

**Comment by akien-mga** [comment]
Yeah the command line interface is a bit hackish :)

Fixed in e3a1bf247 by adding `-export_debug` to the help output.

**Comment by akien-mga** [comment]
(Note that on master it's `--export-debug`.)
